### PR TITLE
Rmp comscore

### PR DESCRIPTION
**Pull request by BjornTwachtmann**
The infinite joy of comscore.

### CODE REVIEW

**Comment by programmes-developers** [comment]
Can one of the admins verify this patch?

File: tests/DataFixtures/PagesService/OptionsFixture.php [hunk 1]
```diff
@@ -127,6 +127,95 @@ public static function radioFour()
...
+                'comments_episodes_enabled' => false,
+                'comments_episodes_id' => '',
+                'playlister_popularity_enabled' => 'no',
+                'bbc_site' => '',
+                'recipes_enabled' => '',
+                'live_stream_id' => '',
+                'live_stream_heading' => '',
+                'control-17' => '',
+                'projectId' => 'progs-worldservice',
+                'navigation_links' => [
```
Comment (BPScott) [inline tests/DataFixtures/PagesService/OptionsFixture.php]: I think we could blank out the navigation_links and promoted_programmes options to make the fixture a bit shorter without affecting anything but that's not a big deal

File: gulpfile.js [hunk 2]
```diff
@@ -37,9 +38,13 @@ gulp.task('js', ['js:clean'], function () {
...
             "respimg": "../../../node_modules/lazysizes/plugins/respimg/ls.respimg",
             "lazysizes": "../../../node_modules/lazysizes/lazysizes-umd",
-            "eqjs": "../../../node_modules/eq.js/build/eq"
+            "eqjs": "../../../node_modules/eq.js/build/eq",
+            "comscorews" : "../../../vendor/bbc-rmp/comscore/js-modules/comscorews",
+            "rmpcomscore/base" : "../../../vendor/bbc-rmp/comscore/js-modules/base",
+            "orb/cookies": "empty:"
         },
         // @TODO do not optimise on sandbox
+        //"optimize": "none",
```
Comment (votemike) [inline gulpfile.js]: What's this?

Comment (BjornTwachtmann) [reply gulpfile.js]: It's how you tell it not to minify your JS.

File: src/ValueObject/ComscoreAnalyticsLabels.php [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+namespace App\ValueObject;
+
+use BBC\ProgrammesPagesService\Domain\Entity\CoreEntity;
+use BBC\ProgrammesPagesService\Domain\Entity\Service;
+use RMP\Comscore\Comscore;
+
+class ComscoreAnalyticsLabels
+{
+    private $comscore;
```
Comment (votemike) [inline src/ValueObject/ComscoreAnalyticsLabels.php]: What's a Comscore?

File: src/ValueObject/ComscoreAnalyticsLabels.php [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+    public function getComscore(): ?Comscore
+    {
+        return $this->comscore;
+    }
+
+    private function setComscore(CosmosInfo $cosmosInfo, IstatsAnalyticsLabels $istatsAnalyticsLabels, string $canonicalUrl)
+    {
+        $labels = $istatsAnalyticsLabels->getLabels();
+        $data = array(
```
Comment (votemike) [inline src/ValueObject/ComscoreAnalyticsLabels.php]: Does the PHPCS allow longform array notation?

Comment (BjornTwachtmann) [reply src/ValueObject/ComscoreAnalyticsLabels.php]: Apparently it does. I should probably read code more thoroughly before I paste it though :)

File: src/ValueObject/ComscoreAnalyticsLabels.php (hunk 3 above)
Comment (BjornTwachtmann) [reply src/ValueObject/ComscoreAnalyticsLabels.php]: The BBC Worldwide analytics system

File: gulpfile.js (hunk 2 above)
Comment (votemike) [reply gulpfile.js]: I mean, why is it here if it is commented out?

File: src/ValueObject/ComscoreAnalyticsLabels.php (hunk 4 above)
Comment (BPScott) [reply src/ValueObject/ComscoreAnalyticsLabels.php]: We ain't in php5.3 world anymore :o

File: gulpfile.js (hunk 2 above)
Comment (BjornTwachtmann) [reply gulpfile.js]: So that I remember this is how to disable minification and eventually set up something to not do this by default on the sandbox.

Comment (votemike) [reply gulpfile.js]: Move the config to a separate file. Then load the config based on env?

Comment (BjornTwachtmann) [reply gulpfile.js]: If that would work, please do so. I have spent enough time swearing at Gulp/Require for this week but it would be useful.

Comment (votemike) [reply gulpfile.js]: Created a ticket https://jira.dev.bbc.co.uk/browse/PROGRAMMES-5845

**Pull request merged**